code/modules/mob/living/simple_animal/hostile/headcrab.dm

```diff
@@ -47,9 +47,13 @@
 			if(C.status_flags & XENO_HOST)
 				src << "<span class='userdanger'>A foreign presence repels us from this body. Perhaps we should try to infest another?</span>"
 				return
+			if(getorgan(/obj/item/organ/internal/body_egg/changeling_egg))
+				src << "<span class='userdanger'>One of us has already planted thier seed in this body! Perhaps we should try to infest another?</span>"
+				return
 			Infect(target)
 			src << "<span class='userdanger'>With our egg laid, our death approaches rapidly...</span>"
 			spawn(100)
+				egg_lain = 0
```

Comment:
Why are you saying there is no egg when you kill the crab, surely the egg disappears when it's released? https://github.com/ArcLumin/Hippie-Rebase/blob/cf0d69d310e218d4c242bc900decdbdffe85c677/code/modules/mob/living/simple_animal/hostile/headcrab.dm#L74 - if so, perhaps you should set egg_lain in the referenced code?